azure-pipelines.dfe.yml

```diff
@@ -107,22 +103,50 @@ jobs:
           command: 'publish'
           publishWebProjects: false
           projects: '**/GovUk.Education.ExploreEducationStatistics.Content.Api.csproj'
-          arguments: '--self-contained true -r win-x64 --configuration $(BuildConfiguration) --output $(build.artifactstagingdirectory)/public-api'
+          arguments: '--self-contained true -r win-x64 --configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)/public-api'
           zipAfterPublish: True
 
-      - task: 'PublishPipelineArtifact@0'
-        displayName: 'Publish Public APIs Artifacts'
+      - task: 'DotNetCoreCLI@2'
+        displayName: 'Publish Public API - API project'
+        inputs:
+          command: 'publish'
+          publishWebProjects: false
+          projects: '**/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.csproj'
+          arguments: '-r linux-musl-x64 --configuration $(BuildConfiguration) --output $(Build.ArtifactStagingDirectory)/public-api/api'
+          zipAfterPublish: False
+
+      - task: Docker@2
+        displayName: 'Build Public API - API Docker image'
+        condition: eq(variables.IsBranchDeployable, true)
+        inputs:
+          containerRegistry: 's101d-datahub-spn-ees-dfe-gov-uk-docker-managed-service-connection'
+          repository: 'ees-public-api/api'
+          command: 'build'
+          Dockerfile: 'docker/public-api/api/Dockerfile'
+          buildContext: '$(Build.ArtifactStagingDirectory)/public-api/api/GovUk.Education.ExploreEducationStatistics.Public.Data.Api'
+          tags: $(Build.BuildNumber)
+          arguments: '--build-arg BUILD_BUILDNUMBER=$(Build.BuildNumber)'
```

Comment:
Just leaving a small comment (hope you don't mind - find this stuff interesting) - some good arguments I've seen passed to docker images like this are: who deployed it (can be sourced via git info I believe), age of the deploy (using a timestamp). Good thing about this is you can expose this info via an endpoint (i.e. `/api/v1/version`) so you can quickly debug stuff if things go haywire in prod

Reply:
Also, found the setting `addPipelineData` in the Docker build task which does a lot of this for free for you :)

Reply:
Ah wicked, just thought: with it being a public api if you end up using semver versioning you could expose the version too